lib/modules/manager/npm/artifacts.ts

```diff
@@ -0,0 +1,91 @@
+import { logger } from '../../../logger';
+import { exec } from '../../../util/exec';
+import type { ExecOptions } from '../../../util/exec/types';
+import { readLocalFile } from '../../../util/fs';
+import { regEx } from '../../../util/regex';
+import type { UpdateArtifact, UpdateArtifactsResult } from '../types';
+
+// eg. 8.15.5+sha256.4b4efa12490e5055d59b9b9fc9438b7d581a6b7af3b5675eb5c5f447cee1a589
+const versionWithHashRegString = '^(?<version>.*)\\+(?<hash>.*)';
+
+// Execute 'corepack use' command for npm manager updates
+// This step is necessary because Corepack recommends attaching a hash after the version
+// The hash is generated only after running 'corepack use' and cannot be fetched from the npm registry
+export async function updateArtifacts({
```

Comment:
@rarkins @secustor we should refactor pm manager to use only this function instead of the  current special behavior